hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -765,6 +774,7 @@ protected Object putInternal(Data key, Object value, long ttl, boolean loadFromS
         if (record == null) {
             record = createRecord(value, ttl, now);
             storage.put(key, record);
+            eventJournal.writeAddEvent(mapContainer.getObjectNamespace(), partitionId, record.getKey(), record.getValue());
```

Comment:
shouldn't we add to journal before `storage#put`? For example, if we have a db backed ring buffer, it seems the true way is first to try insert into db then add to map.